stability-tests/mempool-limits/transactions.go

```diff
@@ -0,0 +1,186 @@
+package main
+
+import (
+	"encoding/hex"
+	"github.com/kaspanet/go-secp256k1"
+	"github.com/kaspanet/kaspad/app/appmessage"
+	"github.com/kaspanet/kaspad/domain/consensus/model/externalapi"
+	"github.com/kaspanet/kaspad/domain/consensus/utils/consensushashing"
+	"github.com/kaspanet/kaspad/domain/consensus/utils/constants"
+	"github.com/kaspanet/kaspad/domain/consensus/utils/subnetworks"
+	"github.com/kaspanet/kaspad/domain/consensus/utils/txscript"
+	utxopkg "github.com/kaspanet/kaspad/domain/consensus/utils/utxo"
+	"github.com/kaspanet/kaspad/domain/dagconfig"
+	"github.com/kaspanet/kaspad/infrastructure/network/rpcclient"
+	"github.com/kaspanet/kaspad/stability-tests/common/mine"
+	"github.com/kaspanet/kaspad/util"
+	"strings"
+)
+
+const (
+	payAddress                       = "kaspasim:qr79e37hxdgkn4xjjmfxvqvayc5gsmsql2660d08u9ej9vnc8lzcywr265u64"
+	payAddressPrivateKey             = "0ec5d7308f65717f3f0c3e4d962d73056c1c255a16593b3989589281b51ad5bc"
+	fundingCoinbaseTransactionAmount = 1000
+	outputsPerTransaction            = 3
+	transactionFee                   = 1000
+	coinbaseMaturity                 = 100
+)
+
+var (
+	payAddressKeyPair     = decodePayAddressKeyPair()
+	payToPayAddressScript = buildPayToPayAddressScript()
+
+	fundingCoinbaseTransactions = make([]*externalapi.DomainTransaction, fundingCoinbaseTransactionAmount)
+)
+
+func generateFundingCoinbaseTransactions(rpcClient *rpcclient.RPCClient) {
+	// Generate one coinbase transaction for its side effect:
+	// the block containing it accepts the empty genesis coinbase
+	generateCoinbaseTransaction(rpcClient)
+
+	log.Infof("Generating funding coinbase transactions")
+	for i := 0; i < fundingCoinbaseTransactionAmount; i++ {
+		fundingCoinbaseTransactions[i] = generateCoinbaseTransaction(rpcClient)
+	}
+
+	log.Infof("Maturing funding coinbase transactions")
+	for i := 0; i < coinbaseMaturity; i++ {
+		generateCoinbaseTransaction(rpcClient)
+	}
+}
+
+func submitAnAmountOfTransactionsToTheMempool(
+	rpcClient *rpcclient.RPCClient, amountToSubmit int, ignoreOrphanRejects bool) {
+
+	log.Infof("Generating %d transactions", amountToSubmit)
+	transactions := make([]*externalapi.DomainTransaction, 0)
+	for len(transactions) < amountToSubmit {
+		var coinbaseTransaction *externalapi.DomainTransaction
+		coinbaseTransaction, fundingCoinbaseTransactions = fundingCoinbaseTransactions[0], fundingCoinbaseTransactions[1:]
```

Comment:
fundingCoinbaseTransactions should be passed around, not a global.

Otherwise it's really hard to understand where is it coming from, and in case it's not pre-filled correctly, very hard to find what happened.